src/main/java/com/palantir/docker/compose/service/InlineDockerServiceBuilder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.docker.compose.service;
+
+import com.palantir.docker.compose.connection.waiting.HealthCheck;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+
+public class InlineDockerServiceBuilder {
+
+    private final String imageName;
+    private final String serviceName;
+    private ComposePortMappings ports = new ComposePortMappings();
+
+    public InlineDockerServiceBuilder(String imageName, String serviceName) {
+        this.imageName = imageName;
+        this.serviceName = serviceName;
+    }
+
+    public InlineDockerServiceBuilder withPortMapping(int port) {
+        ports = ports.withPort(new ComposePortDefinition(port));
+        return this;
+    }
+
+    public InlineDockerServiceBuilder withPortMapping(int internalPort, int externalPort) {
+        ports = ports.withPort(new ComposePortDefinition(internalPort, externalPort));
+        return this;
+    }
+
+    public DockerService build() {
+        File dockerComposeFile = buildDockerComposeFile();
+        return DockerService.fromDockerCompositionFile(dockerComposeFile);
+    }
+
+    public DockerService withHealthCheck(HealthCheck healthCheck) {
+        DockerService service = build();
```

Comment:
why the random build on this method? Why is there an assumption that this gets called last?